examples/AIPL14/SMTP/Problem3.links

```diff
@@ -20,52 +20,53 @@ typename SMTPServer =
                                  send |&] |+],
        QUIT:EndBang |&];
 
-
-sig mailServer : (SMTPServer) ~> EndBang
-fun mailServer(s) {
-  offer(s) {
-    case MAIL(s) ->
-      var (address, s) = receive(s);
-      println("S: received address as FROM: " ^^ address);
-      if (valid(address)) {
-        var s = send("250 OK", select ACCEPT s);
-        rcpt(s)
-      } else {
-        var s = send("Invalid e-mail", select REJECT s);
-        mailServer(s)
-      }
-    case QUIT(s) ->
-      s
-  }
-}
-
-fun rcpt(s) {
-  offer(s) {
-    case RCPT(s) ->
-      var (address, s) = receive(s);
-      println("S: received address as TO: " ^^ address);
-      if (valid(address)) {
-        var s = send("250 OK", select ACCEPT s);
-        rcpt(s)
-      } else {
-        var s = send("Invalid e-mail", select REJECT s);
-        rcpt(s)
-      }
-    case DATA(s) ->
-      var (message, s) = receive(s);
-      println("S: received message: " ^^ message);
-      var s = send("250 OK", s);
-      mailServer(s)
-  }
-}
-
 # Returns true, if the given String is a valid e-mail address.
 # The bit before @ has to be a single word!
 sig valid : (String) ~> Bool
 fun valid(address) {
   address =~ /.[a-z0-9]+.@.[a-z]+.[a-z]+/ # Regex to match 1+characters/numbers @ 1+characters . 1+characters
 }
 
+mutual {
```

Comment:
The sheer number of changes due solely to indentation (inside `mutual` blocks) makes me very sad.  There are also a lot of inevitable changes due to the need to reorder things that are not mutually recursive.